File: src/vn/com/onesoft/livetube/io/core/PingThreadManager.java
package vn.com.onesoft.livetube.io.core;

import vn.com.onesoft.livetube.io.message.cs.CSPing;

public class PingThreadManager {

	private long idConnection = 0;

	private static PingThreadManager _instance;

	public static PingThreadManager getInstance() {
		if (_instance == null) {
			_instance = new PingThreadManager();
			_instance.start();
		}
		return _instance;
	}

	private void start() {
		new Thread(new Runnable() {

			public void run() {
				while (true) {
					ConnectionManager.getInstance().write(new CSPing());

					try {
						Thread.sleep(5000);
					} catch (InterruptedException e) {

					}
				}
			}
		}).start();
	}

}

File: client/ios/BigFoxClient/BigFoxClient/core/session/PingThreadManager.m
//
//  PingThreadManager.m
//  BigFoxClient
//
//  Created by QuyenNX on 9/4/15.
//  Copyright (c) 2015 QuyenNX. All rights reserved.
//

#import "PingThreadManager.h"
#import "CSPing.h"
#import "ConnectionManager.h"

@implementation PingThreadManager
static PingThreadManager* _instance;

+ (PingThreadManager*) getInstance {
    if (_instance == nil) {
        _instance = [[PingThreadManager alloc]init];
        [_instance start];
    }
    return _instance;
}

- (void) start {
    dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
        while (true) {
            [[ConnectionManager getInstance] write:[[CSPing alloc] init]];
            sleep(5);
        }
    });
}
@end
